hotxlfp/grammarparser/lexer.py

```diff
@@ -52,8 +47,13 @@ def t_STRING(t):
     return t
 
 
+available_functions = dispatcher._registry_.keys()
+function_options = '|'.join(f'({re.escape(function)})' for function in available_functions)
+lookahead = '(?=\s*[(])'
+function_regex = f"({function_options}){lookahead}"
```

Comment:
using a lookahead here seems fine. if we run into problems with the regex, you should consider splitting this into two tokens FUNCTION and FUNCTION_CALL - it might be easier to build the parser that way